src/dotnet/CarbonAware.DataSources.Registration/Configuration/ServiceCollectionExtensions.cs

```diff
@@ -0,0 +1,26 @@
+using Microsoft.Extensions.DependencyInjection;
+using CarbonAware.DataSources.Json.Configuration;
+using CarbonAware.DataSources.WattTime.Configuration;
+
+namespace CarbonAware.DataSources.Configuration;
+public static class ServiceCollectionExtensions
+{
+    public static void AddDataSourceService(this IServiceCollection services, DataSourceType pType)
```

Comment:
```suggestion
    public static void AddDataSourceService(this IServiceCollection services, DataSourceType dataSourceType)
```

and elsewhere as appropriate